File: src/org/r3/db/mongo/quiz/quizquestioncollection/QuizQuestionCollectionDTO.java
//
// Understanding Department - 2016 
//
//
// Mongo - TODO: Readme Section.
//




package org.r3.db.mongo.quiz.quizquestioncollection;

import org.r3.db.MongoDbSchemaInfo;
import org.mongodb.morphia.annotations.Entity;

/*
 * Please Define in MongoDbSchemaInfo: public static final String MongoDbQuizQuestionCollectionName = "r3_quizquestioncollection";
 */
@Entity(value=MongoDbSchemaInfo.MongoDbQuizQuestionCollectionName, noClassnameStored=true)
public class QuizQuestionCollectionDTO extends QuizQuestionCollectionDTOBase{
	public QuizQuestionCollectionDTO()
    {
        super();
    }

	public QuizQuestionCollectionDTO clearSensitiveInformation()
	{
		if (answers != null && answers.size() > 0)
		{
			for(AnswersDTO dto : answers )
			{
				dto.setValid(false);
			}
		}
		
		return this;
		
	}
	
	public AnswersDTO getAnswer(String answerId)
	{
		if (answers != null)
		{
			for(QuizQuestionCollectionDTOBase.AnswersDTO a : answers)
			{
				if (a.getAnswerid().equalsIgnoreCase(answerId))
					return a;
			}
		}
		
		return null;
	}
}

File: WebContent/org/r3/db/mongo/quiz/quizquestioncollection/QuizQuestionCollectionDTO.js
//
// Understanding Department - 2016 
//
//
// Mongo - TODO: Readme Section.
//


jQuery.r3Utils.namespace("org.r3.db.mongo.quiz.quizquestioncollection");

org.r3.db.mongo.quiz.quizquestioncollection.QuizQuestionCollectionDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
       this.dtrowclass   = "grade_quizquestioncollection";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
       this.questionid = aDataBean.questionid || null;
       this.quizid = aDataBean.quizid || null;
       this.text = aDataBean.text || null;
       this.qlevel = aDataBean.qlevel || null;
       this.answers = aDataBean.answers || null;
	 }
       
});

 org.r3.db.mongo.quiz.quizquestioncollection.AnswersDTO = Class.extend(
 {
     init: function(aDataBean)
	 {
        this.answerid = aDataBean.answerid || null;
        this.text = aDataBean.text || null;
        this.valid = aDataBean.valid || null;
        this.weight = aDataBean.weight || null;
	 }	   
 });   
  
